plugins/async-node/ios/Sources/AsyncNodePlugin.swift

```diff
@@ -7,124 +7,149 @@
      */
     public convenience init(plugins: [JSBasePlugin] = [AsyncNodePluginPlugin()], _ handler: @escaping AsyncHookHandler) {
-
+        
         self.init(fileName: "AsyncNodePlugin.native", pluginName: "AsyncNodePlugin.AsyncNodePlugin")
         self.asyncHookHandler = handler
         self.plugins = plugins
     }
-
+    
+    /**
+     Converts a given `ReplacementNode` to a `JSValue` that can be used in the JavaScript context.
+     
+     - Parameters:
+     - replacementNode: The `ReplacementNode` to be converted.
+     - context: The `JSContext` in which the `JSValue` will be used.
+     - Returns: A `JSValue` representing the given `ReplacementNode`, or `nil` if the conversion fails.
+     */
+    func convertReplacementNodeToJSValue(_ replacementNode: ReplacementNode, context: JSContext) -> JSValue? {
+        switch replacementNode {
+        case .encodable(let encodable):
+            let encoder = JSONEncoder()
+            do {
+                let res = try encoder.encode(encodable)
+                return context.evaluateScript("(\(String(data: res, encoding: .utf8) ?? ""))") as JSValue
+            } catch {
+                return nil
+            }
+        case .concrete(let jsValue):
+            return jsValue
+        }
+    }
+    
+    /**
+     Converts a given `AsyncNodeHandlerType` to a `JSValue` that can be used in the JavaScript context.
+     
+     - Parameter replacementNode: The `AsyncNodeHandlerType` to be converted.
+     - Returns: A `JSValue` representing the given `AsyncNodeHandlerType`, or `nil` if the conversion fails.
+     */
+    
+    func replacementNodeToJSValue(_ replacementNode: AsyncNodeHandlerType) -> JSValue? {
+        guard let context = context else {
+            return JSValue()
+        }
+        switch replacementNode {
+        case .multiNode(let replacementNodes):
+            let jsValueArray = replacementNodes.compactMap { node in
+                switch node {
+                    
+                case .encodable(let encodable):
+                    return convertReplacementNodeToJSValue(.encodable(encodable), context: context)
+                case .concrete(let jsValue):
+                    return convertReplacementNodeToJSValue(.concrete(jsValue), context: context)
+                    
+                }
+            }
+            return context.objectForKeyedSubscript("Array").objectForKeyedSubscript("from").call(withArguments: [jsValueArray])
+            
+        case .singleNode(let replacementNode):
+            switch replacementNode {
+                
+            case .encodable(let encodable):
+                return convertReplacementNodeToJSValue(.encodable(encodable), context: context)
+            case .concrete(let jsValue):
+                return convertReplacementNodeToJSValue(.concrete(jsValue), context: context)
+                
+            }
```

Comment:
this switch is not needed, since `convertReplacementNodeToJSValue` already switches inside